src/Network.py

```diff
@@ -52,10 +53,32 @@ def generate_nodes(cls,n_nodes=2,topology='FULL'):
                                       missing)
                 for node in nodes:
                     filtered_nodes = [nodes[edge[1]] for edge in graph.edges(int(node.name))]
-                    # Adding the node itself to the set of nodes for the quorum set
-                    filtered_nodes.append(node)
-                    log.network.debug('Adding nodes %s to the quorum set of Node %s',
-                                      filtered_nodes, node)
-                    node.set_quorum(filtered_nodes)
+
+                    if len(filtered_nodes) > 1:
+                        filter_distribution = len(filtered_nodes)//2 # Filter nodes for Quorums as half to Quorum and half to inner sets
+
+                        quorum_distribution = filtered_nodes[:filter_distribution]
+                        inner_quorum_distribution = filtered_nodes[filter_distribution:]
```

Comment:
so we get the last and first sets from the quorum?

Reply:
When we distribute how many nodes should go to the quorum and to the inner sets, we distribute based on the number of nodes being added to the quorum. I have set this to a 50:50 split so that inner sets can be formed, but it may have to be changed to a different number. We should also discuss this with Matija